lib/elCanvas2.js

```diff
@@ -0,0 +1,807 @@
+/**
+ * Work-in-Progress, take all the code made by Lucky Dee and
+ * - standardize naming
+ * - standardize coding style
+ * - Use javascript prototype for efficiency
+ *
+ * Update by Glenn R. Wichman
+ * Should work identically to original elCanvas EXCEPT
+ * original elCanvas allowed you to put the "new" operator in front of objectType functions
+ * e.g. var r = new el.rect(...);
```

Comment:
just a note on the "new", each canvas should be seperate, if 2 elCanvases are made they should not share the same objects, same custom shapes or anything else, as long as we can still keep new elCanvas and each elCanvas is unique this is fine